src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs

```diff
@@ -373,12 +374,13 @@ private bool CanContinueBatch(ReplicationState state, ref long next)
         {
             if (MissingAttachmentsInLastBatch)
             {
-                state.MissingTxMarkers.Remove(state.LastTransactionMarker);
+                // we do have missing attachments but we haven't gathered yet any of the missing hashes
+                if (state.MissingAttachmentBase64Hashes == null)
```

Comment:
I don't understand how that scenario can be?

Reply:
We are iterating items by etag, documents can be later on, but I do not want to break the batch before that? I think it should not be a case BTW, but prefer to stay on the safe side, I would like to not break the batch to early.

Reply:
attachments will _always_ happen before their document.